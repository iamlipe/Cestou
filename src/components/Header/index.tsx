import React, {memo} from 'react';
import styled from 'styled-components/native';
import {useReduxSelector} from '@/hooks/useReduxSelector';

interface Props {
  welcome?: boolean;
  title?: string;
}

const Header = ({welcome = true, title}: Props) => {
  const user = useReduxSelector(state => state.user.auth);

  return (
    <StyledContainer>
      {welcome ? (
        <StyledTitle>{`Ol√° ${user?.firstName}!`}</StyledTitle>
      ) : (
        <StyledTitle>{title}</StyledTitle>
      )}
    </StyledContainer>
  );
};

const StyledContainer = styled.View`
  width: 100%;
  height: 64px;
  flex-direction: row;
  align-items: center;
  background-color: ${({theme}) => theme.colors.PRIMARY_300};
  padding: 0 5%;
`;

const StyledTitle = styled.Text`
  font-family: ${({theme}) => theme.fonts.SEMIBOLD_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.LARGEST};
  color: ${({theme}) => theme.colors.GRAY_900};
`;

export default memo(Header);
