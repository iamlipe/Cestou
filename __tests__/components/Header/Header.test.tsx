import React from 'react';
import {
  renderWithThemeProvider,
  matchSnapshotWithProvider,
} from '@/config/functions/testing';

import Header from '@/components/Header';

describe('Header', () => {
  test('should render correctly', () => {
    renderWithThemeProvider(<Header />);
  });

  test('should matches snapshot', () => {
    matchSnapshotWithProvider(<Header />);
  });

  test('should render with title', () => {
    const header = renderWithThemeProvider(
      <Header title="Title" welcome={false} />,
    );

    const title = header.getByText('Title');

    expect(title).toBeTruthy();
  });

  test('should render with a welcome message', () => {
    const header = renderWithThemeProvider(<Header />, {
      user: {auth: {firstName: 'Mason'}},
    });

    const welcome = header.queryByText(/olÃ¡ mason!/i);

    expect(welcome).toBeTruthy();
  });

  test('should render type profile', () => {
    const header = renderWithThemeProvider(
      <Header title="Profile" type="profile" welcome={false} />,
      {
        user: {auth: {firstName: 'Mason'}},
      },
    );

    const userName = header.queryByText(/mason/i);
    const userPhoto = header.queryByTestId('user-photo');
    const buttonSettings = header.queryByTestId('icon-settings');

    expect(userName).toBeTruthy();
    expect(userPhoto).toBeTruthy();
    expect(buttonSettings).toBeTruthy();
  });
});
