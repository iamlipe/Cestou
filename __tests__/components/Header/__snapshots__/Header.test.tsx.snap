// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should matches snapshot 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#6CCD91",
          "flexDirection": "row",
          "height": 64,
          "paddingBottom": 0,
          "paddingLeft": "5%",
          "paddingRight": "5%",
          "paddingTop": 0,
          "width": "100%",
        },
      ]
    }
    type="normal"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "space-between",
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#1A1A1A",
              "fontFamily": "SourceSansPro-SemiBold",
              "fontSize": 47,
              "marginLeft": 10,
            },
          ]
        }
      >
        Ol√° undefined!
      </Text>
    </View>
  </View>
</View>
`;
